### ISSUE TITLE
Implement __aenter__ and __aexit__ in Client class for use with context manager

### ISSUE THREAD

**Issue by cwrneiro** [mathsman5133/coc.py#121]
This simple workaround is already enough to work with asynchronous context managers:

```python
class Client(coc.Client):
	async def __aenter__(self):
		self.__init__()
		return self

	async def __aexit__(self, *args):
		await self.http.close()
```

It can be easily implemented into the official code for easier use of the Client class, without the need to close after using it (also making it more organized):
```python
async def main():
	async with coc.Client() as client:
		# do stuff...

if __name__ == "__main__":
	asyncio.run(main())
```
versus:
```python
async def main():
	client =  coc.Client():
	# do stuff...
	client.close()

if __name__ == "__main__":
	asyncio.run(main())
```

**Comment by majordoobie** [comment]
Great work! This was something that the creator wanted to add. Did you run all the examples with this update?

**Comment by cwrneiro** [comment]
Thanks! I've only tried with basic scripts, but it should work with almost anything as it uses the class own __init__() and http.close() methods

**Comment by doluk** [comment]
If you create a pr, we can test and merge it!

**Comment by majordoobie** [comment]
Thank you for the suggestion. Context manager is now available in v2.2.0
